x-pack/legacy/plugins/siem/public/components/url_state/helpers.ts

```diff
@@ -71,50 +74,60 @@ export const replaceQueryStringInLocation = (location: Location, queryString: st
   }
 };
 
-export const getUrlType = (pathname: string): UrlStateType => {
-  const removeSlash = pathname.replace(/\/$/, '');
-  const trailingPath = removeSlash.match(/([^\/]+$)/);
-  if (trailingPath !== null) {
-    if (trailingPath[0] === 'hosts' || pathname.match(/^\/hosts\/.+$/) != null) {
-      return 'host';
-    } else if (trailingPath[0] === 'network' || pathname.match(/^\/network\/.+$/) != null) {
-      return 'network';
-    } else if (trailingPath[0] === 'overview') {
-      return 'overview';
-    } else if (trailingPath[0] === 'timelines') {
-      return 'timeline';
-    }
+export const getUrlType = (pageName: string): UrlStateType => {
```

Comment:
It seems this does the trick.
```
  return getOr(SiemPageName.overview, pageName, SiemPageName);
```

To do this, we have some `urlKey` to update: Change `host` to `hosts`

1. https://github.com/elastic/kibana/blob/90a45590664e7bb99e044d98d8ccfbf1611f4588/x-pack/legacy/plugins/siem/public/components/navigation/tab_navigation/index.tsx#L116
2. https://github.com/elastic/kibana/blob/90a45590664e7bb99e044d98d8ccfbf1611f4588/x-pack/legacy/plugins/siem/public/components/url_state/types.ts#L19
3. https://github.com/elastic/kibana/blob/90a45590664e7bb99e044d98d8ccfbf1611f4588/x-pack/legacy/plugins/siem/public/components/url_state/types.ts#L28
4. https://github.com/elastic/kibana/blob/90a45590664e7bb99e044d98d8ccfbf1611f4588/x-pack/legacy/plugins/siem/public/pages/home/home_navigations.tsx#L43
5. https://github.com/elastic/kibana/blob/90a45590664e7bb99e044d98d8ccfbf1611f4588/x-pack/legacy/plugins/siem/public/pages/hosts/hosts_navigations.tsx#L48-L84
6. https://github.com/elastic/kibana/blob/90a45590664e7bb99e044d98d8ccfbf1611f4588/x-pack/legacy/plugins/siem/public/pages/hosts/hosts_navigations.tsx#L92-L116

Given that now both `id` and `urlKey` are both align with `SiemPageName`, so replace `urlKey` with `id` is probably the next thing I would like to do.

Reply:
yes, I was thinking about it but for some reason, I was not ready to change it everywhere.
I still think that urlKey is different because this help to link back to the redux store, Maybe I can show you and we can talk over zoom about it